apps/app_voicemail.c

```diff
@@ -13394,6 +13551,104 @@ static int manager_list_voicemail_users(struct mansession *s, const struct messa
 	return RESULT_SUCCESS;
 }
 
+static int manager_get_mailbox_summary(struct mansession *s, const struct message *m)
+{
+	struct ast_vm_user *vmu = NULL;
+	const char *id = astman_get_header(m, "ActionID");
+	char actionid[128];
+	struct ast_vm_user svm;
+	int nummessages;
+
+	const char *context = astman_get_header(m, "Context");
+	const char *mailbox = astman_get_header(m, "Mailbox");
+
+	if ((ast_strlen_zero(context) || ast_strlen_zero(mailbox))) {
+		astman_send_error(s, m, "Need 'Context' and 'Mailbox' parameters.");
+		return RESULT_SUCCESS;
+	}
+
+	actionid[0] = '\0';
+	if (!ast_strlen_zero(id)) {
+		snprintf(actionid, sizeof(actionid), "ActionID: %s\r\n", id);
+	}
+
+	/* find user */
+	memset(&svm, 0, sizeof(svm));
+	vmu = find_user(&svm, context, mailbox);
+	if (!vmu) {
+		/* could not find it */
+		astman_send_ack(s, m, "There is no voicemail user of the given info.");
+		return RESULT_SUCCESS;
+	}
+
+	astman_send_listack(s, m, "Voicemail box detail will follow", "start");
+
+	/* walk through user's mailbox and append info for each message */
+	nummessages = append_vmbox_info_astman(s, vmu, "VoicemailBoxDetail", actionid);
+	if(nummessages < 0) {
+		ast_log(LOG_ERROR, "Could not append voicemail box info.");
+		nummessages = 0;
+	}
+	free_user(vmu);
+
+	astman_send_list_complete_start(s, m, "VoicemailBoxDetailComplete", nummessages);
+	astman_send_list_complete_end(s);
+
+	return RESULT_SUCCESS;
+}
+
+static int manager_voicemail_move(struct mansession *s, const struct message *m)
+{
+	const char *mailbox = astman_get_header(m, "Mailbox");
+	const char *context = astman_get_header(m, "Context");
+	const char *from_folder = astman_get_header(m, "Folder");
+	const char *id[] = { astman_get_header(m, "ID") };
+	const char *to_folder = astman_get_header(m, "ToFolder");
+
```

Comment:
This should enforce any required headers so the AMI user gets a more detailed response